libs/langchain/langchain/cache.py

```diff
@@ -470,21 +451,63 @@ def _get_gptcache(self, llm_string: str) -> Any:
             _gptcache = self._new_gptcache(llm_string)
         return _gptcache
 
-    def lookup(self, prompt: str, llm_string: str) -> Optional[RETURN_VAL_TYPE]:
+    def lookup(self, prompt: str, llm_string: str) -> Optional[List[ChatGeneration]]:
         """Look up the cache data.
-        First, retrieve the corresponding cache object using the `llm_string` parameter,
-        and then retrieve the data from the cache based on the `prompt`.
+
+        Args:
+            prompt (str): The prompt to search for.
+            llm_string (str): The language model string.
+
+        Returns:
+            Optional[List[ChatGeneration]]: \
+                List of ChatGeneration instances if found, else None.
         """
-        from gptcache.adapter.api import get
+        try:
+            from gptcache.adapter.api import get
+        except ImportError:
+            raise ImportError(
+                "Could not import gptcache python package. "
+                "Please install it with `pip install gptcache`."
+            )
+        gpt_cache = self.gptcache_dict.get(llm_string)
```

Comment:
@kdcokenny we are introducting a bug here. Please check this - https://github.com/langchain-ai/langchain/pull/8417